agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/exporter/AzureMonitorMetricsDataTest.java

```diff
@@ -0,0 +1,313 @@
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.metrics.export.MetricReaderFactory;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricExporter;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AzureMonitorMetricsDataTest {
+
+  private Meter meter;
+  private InMemoryMetricExporter inMemoryMetricExporter;
+
+  @BeforeEach
+  public void setup() {
+    inMemoryMetricExporter = InMemoryMetricExporter.create();
+    MetricReaderFactory metricReaderFactory =
+        PeriodicMetricReader.newMetricReaderFactory(inMemoryMetricExporter);
+    SdkMeterProvider meterProvider =
+        SdkMeterProvider.builder().registerMetricReader(metricReaderFactory).build();
+
+    OpenTelemetry openTelemetry =
+        OpenTelemetrySdk.builder().setMeterProvider(meterProvider).build();
+
+    meter =
+        openTelemetry
+            .meterBuilder("AzureMonitorMetricsDataTest")
+            .setInstrumentationVersion("1.0.0")
+            .build();
+  }
+
+  @AfterEach
+  public void cleanup() {
+    inMemoryMetricExporter.reset();
+  }
+
+  @Test
+  public void testDoubleCounter() throws InterruptedException {
+    DoubleCounter counter = meter.counterBuilder("testDoubleCounter").ofDoubles().build();
+    counter.add(3.1415);
+
+    Thread.sleep(90 * 1000); // wait 90 seconds
+
+    List<MetricData> metricDatas = inMemoryMetricExporter.getFinishedMetricItems();
+    assertThat(metricDatas.size()).isEqualTo(1);
+
+    MetricData metricData = metricDatas.get(0);
+    for (PointData pointData : metricData.getData().getPoints()) {
+      MetricTelemetryBuilder builder = TelemetryClient.createForTest().newMetricTelemetryBuilder();
+      AzureMonitorMetricsData.populatePointAndProperties(builder, metricDatas.get(0), pointData);
+      MetricsData metricsData = (MetricsData) builder.build().getData().getBaseData();
+      assertThat(metricsData.getMetrics().size()).isEqualTo(1);
+      assertThat(metricsData.getMetrics().get(0).getValue()).isEqualTo(3.1415);
+    }
+
+    assertThat(metricData.getType()).isEqualTo(DOUBLE_SUM);
+    assertThat(metricData.getName()).isEqualTo("testDoubleCounter");
+  }
+
+  @Test
+  public void testDoubleGauge() throws InterruptedException {
+    meter
+        .gaugeBuilder("testDoubleGauge")
+        .setDescription("the current temperature")
+        .setUnit("C")
+        .buildWithCallback(
+            m -> {
+              m.record(20.0, Attributes.of(AttributeKey.stringKey("thing"), "engine"));
+            });
+
+    Thread.sleep(90 * 1000); // wait 90 seconds
```

Comment:
let's set the metric interval smaller so the tests don't take so long, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/31f5734be6946b074e41466019382a3cb0e18422/testing-common/src/main/java/io/opentelemetry/instrumentation/testing/LibraryTestRunner.java#L60-L62

Reply:
well, opentelemetry sdk metrics take at least 1 min.

Reply:
that's configurable (see link above)

I pushed change to your branch

Reply:
i forgot that part.. cool thanks.